include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -27,7 +27,7 @@ void from_json(const BasicJsonType& j, typename std::nullptr_t& n)
 {
     if (JSON_HEDLEY_UNLIKELY(!j.is_null()))
     {
-        JSON_THROW(type_error::create(302, "type must be null, but is " + std::string(j.type_name())));
+        JSON_THROW(type_error::create(302, j.diagnostics() + "type must be null, but is " + std::string(j.type_name())));
```

Comment:
After seeing this several times, I'm wondering if `type_error::create` should instead take a second parameter of type `BasicJsonType` so that you can't forget to do the `j.diagnostics() + `.

Reply:
Looks like there are many more uses than the ones here that add the diagnostics, so perhaps not.

Reply:
I was thinking about this, too. I first wanted to get things "right" before cleaning up. You are definitely right!

Reply:
Done.